### PR TITLE
Add legacy slider offsets to osu!/osu!catch

### PR DESCRIPTION
**Pull request by smoogipoo**
Prereqs:
- [x] https://github.com/ppy/osu/pull/2635

---

Adds support for arguably "incorrect" magic offsets which are added to certain hitobjects in stable as hacky workarounds to issues over the years. Will only apply to legacy converted beatmaps, but help bring them in line as far as conversion and difficulty calculation.

### CODE REVIEW

**Comment by smoogipoo** [comment]
Should resolve any minor inaccuracies in SR calculation due to correct ticks now too :)

File: osu.Game.Rulesets.Osu/Objects/Slider.cs [hunk 1]
```diff
@@ -91,6 +94,9 @@ protected override void CreateNestedHitObjects()
             createSliderEnds();
             createTicks();
             createRepeatPoints();
+
+            if (LegacyLastTickOffset != null)
+                TailCircle.StartTime = Math.Max(StartTime + Duration / 2, TailCircle.StartTime - LegacyLastTickOffset.Value);
```
Comment (peppy) [inline osu.Game.Rulesets.Osu/Objects/Slider.cs]: where does the /2 come from here?

Comment (smoogipoo) [reply osu.Game.Rulesets.Osu/Objects/Slider.cs]: This is just what stable does

**Comment by peppy** [comment]
Rebased for clarity

**Review by peppy — APPROVED** [review]

**Pull request merged**